staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder.go

```diff
@@ -0,0 +1,394 @@
+	}
+
+	return description
+}
+
+// buildRoute returns a RouteBuilder for WebService to consume and builds path in swagger
+func (b *builder) buildRoute(root, path, action, verb string, sample interface{}) *restful.RouteBuilder {
+	var namespaced string
+	if b.namespaced {
+		namespaced = "Namespaced"
+	}
+	route := b.ws.Method(action).
+		Path(root+path).
+		To(func(req *restful.Request, res *restful.Response) {}).
+		Doc(b.descriptionFor(path, verb)).
+		Param(b.ws.QueryParameter("pretty", "If 'true', then the output is pretty printed.")).
+		Operation(verb+namespaced+b.kind+strings.Title(subresource(path))).
+		Metadata(endpoints.ROUTE_META_GVK, metav1.GroupVersionKind{
+			Group:   b.group,
+			Version: b.version,
+			Kind:    b.kind,
+		}).
+		Metadata(endpoints.ROUTE_META_ACTION, strings.ToLower(action)).
+		Produces("application/json", "application/yaml").
+		Returns(http.StatusOK, "OK", sample).
+		Writes(sample)
+
+	if action == "PATCH" {
+		route.Consumes("application/json-patch+json",
+			"application/merge-patch+json",
+			"application/strategic-merge-patch+json")
+	} else {
+		route.Consumes("*/*")
+	}
+
+	switch verb {
+	case "get":
+		// TODO: CRD support for export is still under consideration
+		endpoints.AddObjectParams(b.ws, route, &metav1.GetOptions{})
+	case "list", "deletecollection":
+		endpoints.AddObjectParams(b.ws, route, &metav1.ListOptions{})
+	case "replace", "patch":
+		// TODO: PatchOption added in feature branch but not in master yet
+		endpoints.AddObjectParams(b.ws, route, &metav1.UpdateOptions{})
+	case "create":
+		endpoints.AddObjectParams(b.ws, route, &metav1.CreateOptions{})
+	case "delete":
+		endpoints.AddObjectParams(b.ws, route, &metav1.DeleteOptions{})
+	}
+
+	switch verb {
+	case "create":
+		route.Returns(http.StatusAccepted, "Accepted", sample)
+		route.Returns(http.StatusCreated, "Created", sample)
+	case "delete":
+		route.Returns(http.StatusAccepted, "Accepted", sample)
+	case "replace":
+		route.Returns(http.StatusCreated, "Created", sample)
+	}
+
+	return route
+}
+
+// buildKubeNative builds input schema with Kubernetes' native object meta, type meta and
+// extensions
+func (b *builder) buildKubeNative(schema *spec.Schema) *spec.Schema {
+	// schema.SetProperty("metadata", getDefinition(objectMetaType))
+	schema.SetProperty("metadata", *spec.RefSchema(objectMetaSchemaRef).
+		WithDescription(swaggerPartialObjectMetadataDescriptions["metadata"]))
+	addTypeMetaProperties(schema)
+	schema.AddExtension(endpoints.ROUTE_META_GVK, []map[string]string{
+		{
+			"group":   b.group,
+			"version": b.version,
+			"kind":    b.kind,
+		},
+	})
+
+	return schema
+}
+
+// getDefinition gets definition for given Kubernetes type. This function is extracted from
+// kube-openapi builder logic
+func getDefinition(name string) spec.Schema {
+	definitions := generatedopenapi.GetOpenAPIDefinitions(func(name string) spec.Ref {
+		defName, _ := openapi.NewDefinitionNamer(legacyscheme.Scheme).GetDefinitionName(name)
```

Comment:
we can't use the global scheme here. The apiextensions-apiserver must not know anything about the kube-apiserver.